cloudscale/resource_cloudscale_volume_test.go

```diff
@@ -0,0 +1,270 @@
+					resource.TestCheckResourceAttr(
+						"cloudscale_volume.basic", "server_uuids.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudscaleVolume_Reattach(t *testing.T) {
+	var server1, server2 cloudscale.Server
+	var volume cloudscale.Volume
+
+	rInt1 := acctest.RandInt()
+	rInt2 := acctest.RandInt()
+	rInt3 := acctest.RandInt()
+
+	serverConfig := testAccCheckCloudscaleServerConfig_basic(rInt1)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudscaleVolumeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: serverConfig + "\n" + volumeConfig_attached(rInt2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudscaleServerExists("cloudscale_server.basic", &server1),
+					testAccCheckCloudscaleVolumeExists("cloudscale_volume.basic", &volume),
+					resource.TestCheckResourceAttr(
+						"cloudscale_volume.basic", "server_uuids.#", "1"),
+					assertVolumeAttached(&server1, &volume),
+				),
+			},
+			{
+				Config: serverConfig + "\n" + volumeConfig_reattached_volume(rInt3, rInt2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudscaleServerExists("cloudscale_server.reattach_server", &server2),
+					testAccCheckCloudscaleVolumeExists("cloudscale_volume.basic", &volume),
+					assertVolumeAttached(&server2, &volume),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCloudscaleVolumeDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*cloudscale.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cloudscale_volume" {
+			continue
+		}
+
+		id := rs.Primary.ID
+
+		// Try to find the volume
+		_, err := client.Volumes.Get(context.Background(), id)
+		if err != nil && !strings.Contains(err.Error(), "Not found") {
+			return fmt.Errorf(
+				"Error waiting for volume (%s) to be destroyed: %s",
+				rs.Primary.ID, err)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckCloudscaleVolumeExists(n string, volume *cloudscale.Volume) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Volume ID is set")
+		}
+
+		client := testAccProvider.Meta().(*cloudscale.Client)
+
+		id := rs.Primary.ID
+
+		// Try to find the volume
+		retrieveVolume, err := client.Volumes.Get(context.Background(), id)
+
+		if err != nil {
+			return err
+		}
+
+		if retrieveVolume.UUID != rs.Primary.ID {
+			return fmt.Errorf("Volume not found")
+		}
+
+		*volume = *retrieveVolume
+
+		return nil
+	}
+}
+
+func assertVolumeAttached(server *cloudscale.Server, volume *cloudscale.Volume) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if server.UUID != (*volume.ServerUUIDs)[0] {
+			return fmt.Errorf("Server not properly attached to %s: %v", server.UUID, volume.ServerUUIDs)
```

Comment:
Did you mean "volume not properly attached to %s: %v"? Would it make sense to rephrase it to something like "volume is not attached to %s but to %v"?